psydac/api/fem.py

```diff
@@ -310,11 +310,11 @@ def __init__(self, expr, kernel_expr, domain_h, spaces, *, matrix=None, update_g
 
         #...
         discrete_space   = (trial_space, test_space)
-        space_quad_order = [qo - 1 for qo in get_quad_order(test_space)]
-        quad_order       = [qo + 1 for qo in (quad_order or space_quad_order)]
+        space_nquads = [qo - 1 for qo in get_nquads(test_space)]
+        nquads       = [qo + 1 for qo in (nquads or space_nquads)]
 
         # this doesn't work right now otherwise. TODO: fix this and remove this assertion
-        assert np.array_equal(quad_order, get_quad_order(test_space))
+        assert np.array_equal(nquads, get_nquads(test_space))
```

Comment:
I do not fully understand the reasoning behind these lines of code. Do you?

CC: @saidctb